internal/common/autogen/unmarshal_test.go

```diff
@@ -140,10 +141,10 @@ func TestUnmarshalNestedAllTypes(t *testing.T) {
 		AttrCustomListString:        customtypes.NewListValueUnknown[types.String](ctx),
 		AttrListObj:                 types.ListUnknown(objTypeTest),
 		AttrCustomNestedList: customtypes.NewNestedListValue[modelCustomTypeTest](ctx, []modelCustomTypeTest{
+			modelCustomTypeBasic,
 			{
-				// these attribute values are irrelevant, they will be overwritten with JSON values
-				AttrString:    types.StringValue("different_string"),
-				AttrInt:       types.Int64Value(999),
+				AttrString:    types.StringValue("existing not overwritten"),
```

Comment:
Nice coverage. Something to consider is TestUnmarshalNestedAllTypes is becoming very big. Not high priority but we could consider refactoring to having separate tests for these particular scenarios, with common code so each test definition doesn't imply too much boilerplate